jlab-sensing/soil-power-sensor-firmware
# Integrate Phytos31

**Name/Affiliation/Title**
John, UCSC, protector of code (aka maintainer)

**Purpose of the PR**
Add support for the Phytos31 leaf wetness sensor

**Development Environment**
`Linux spruce 6.8.2-arch2-1 #1 SMP PREEMPT_DYNAMIC Thu, 28 Mar 2024 17:06:35 +0000 x86_64 GNU/Linux`
PlatformIO Core, version 6.1.7
Used ST-Link V3 MINIE

**Test Procedure**
Connect the Phytos31 sensor to the `V+` terminal and ground `V-` to put in single ended configuration. First run the example code to see raw voltage outputs printed over UART. Place hand on Phytos31, the output should change.

In the same configuration run the default env and check for changes on [Dirtviz](https://dirtviz.jlab.ucsc.edu/dashboard) under cell name `demo`.

**Additional Context**
Python protobuf package dependency versions has been updated to hopefully fix #73. Also, the `publish-to-testpypi` is reconfigured to only run on workflow dispatch (#35).

Closes #35
Closes #58 
Closes #73
